### PR TITLE
Get WebIO working on Binder

### PR DESCRIPTION
**Pull request by djsegal**
attempts to resolve: 

+ https://github.com/JuliaGizmos/WebIO.jl/issues/72

### CODE REVIEW

File: src/providers/ijulia_setup.jl [hunk 1]
```diff
@@ -16,15 +16,28 @@ function WebIO.register_renderable(T::Type)
...
-    display(HTML("<script src='/pkg/WebIO/webio/dist/bundle.js'></script>"))
-    display(HTML("<script src='/pkg/WebIO/providers/ijulia_setup.js'></script>"))
+    display(HTML("""
+        <script>
+            var curMatch =
+                window.location.href
+                .match(/(.*)\\/notebooks\\/(.*)\\.ipynb/);
+
+            if ( curMatch ) {
+                \$('head').append('<base href="' + curMatch[1] + '/">');
```
Comment (shashi) [inline src/providers/ijulia_setup.jl]: is `curMatch[1]` correct?

**Comment by djsegal** [comment]
Shoot. Youre right. Ill change it to zero

Its hard going between js and julia

**Comment by djsegal** [comment]
My bad. I lied. Its because i parened the main url leading to /notebooks/

1 is right. Could remove the parens

File: src/providers/ijulia_setup.jl (hunk 1 above)
Comment (djsegal) [reply src/providers/ijulia_setup.jl]: I don't know if it is as correct as much as it works.

The original idea at the time of writing it was that I was trying to parse out the `hostname` too.

----

It does work, though.

Alternatively, we just change:

```julia
                .match(/(.*)\\/notebooks\\/(.*)\\.ipynb/);

            if ( curMatch ) {
                \$('head').append('<base href="' + curMatch[1] + '/">');
```

to

```julia
                .match(/.*\\/notebooks\\/(.*)\\.ipynb/);

            if ( curMatch ) {
                \$('head').append('<base href="' + curMatch[0] + '/">');
```

Comment (shashi) [reply src/providers/ijulia_setup.jl]: It's confusing to me why you want to extract the name of the notebook.

I thought you wanted to set the base url to everything leading up to the `/notebooks/`. I'm probably misunderstanding something.

Comment (djsegal) [reply src/providers/ijulia_setup.jl]: what i said was completely wrong.

+ `curMatch[0]` = the full url (not a partial of the url)

// don't accept this PR for another day or so

**Comment by djsegal** [comment]
@shashi this is all done at this point.

hope i cleared up that one issue. let me know if you have any more thoughts. cheers!

File: src/providers/ijulia_setup.jl (hunk 1 above)
Comment (shashi) [reply src/providers/ijulia_setup.jl]: LOL why not?

Comment (djsegal) [reply src/providers/ijulia_setup.jl]: Sorry. That warning about not accepting PR was pre some other changes.

API for match is: 

+ https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

-------

Quick example for the JS console:

```javascript
> tmpMatch = window.location.href.match(/(.*)\/JuliaGizmos/)
["https://github.com/JuliaGizmos", "https://github.com", index: 0, input: "https://github.com/JuliaGizmos/WebIO.jl/pull/75", groups: undefined]
```

and then

```javascript
> tmpMatch[0]
"https://github.com/JuliaGizmos"
> tmpMatch[1]
"https://github.com"
> tmpMatch[2]
undefined
```

File: src/providers/ijulia_setup.jl [hunk 2]
```diff
@@ -32,8 +32,10 @@ function main()
         </script>
     """))
 
-    display(HTML("<script src='pkg/WebIO/webio/dist/bundle.js'></script>"))
-    display(HTML("<script src='pkg/WebIO/providers/ijulia_setup.js'></script>"))
+    display(HTML("<script class='js-remove-script' src='pkg/WebIO/webio/dist/bundle.js'></script>"))
+    display(HTML("<script class='js-remove-script' src='pkg/WebIO/providers/ijulia_setup.js'></script>"))
+
+    display(HTML("<script class='js-remove-script'>\$('.js-remove-script').closest('.output_area').remove();</script>"))
```
Comment (shashi) [inline src/providers/ijulia_setup.jl]: This will cause reloading the page to lose the JS....

Comment (djsegal) [reply src/providers/ijulia_setup.jl]: Very interesting... I'll just do a padding = 0 trick then!

Comment (djsegal) [reply src/providers/ijulia_setup.jl]: Sorry for trying to slip that in

File: src/providers/ijulia_setup.jl (hunk 1 above)
Comment (shashi) [reply src/providers/ijulia_setup.jl]: Ah, nice! Looks good then. Sorry for not being schooled about how match works. If you revert the scrub output divs commit, I'll hit merge. Appreciate this useful feature!

File: src/providers/ijulia_setup.jl (hunk 2 above)
Comment (djsegal) [reply src/providers/ijulia_setup.jl]: Done!

**Pull request merged**